hw02-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,31 @@
+package homework;
+
+
+import java.util.*;
+import java.util.Map.Entry;
+
+public class CustomerService {
+
+    private final Map<Customer, String> customerMap = new TreeMap<>(Comparator.comparing(Customer::getScores));
+
+    public Entry<Customer, String> getSmallest() {
+        Entry<Customer, String> smallest = customerMap.entrySet().stream().findFirst().orElse(null);
+        try {
+            return new AbstractMap.SimpleEntry<>(new Customer(smallest.getKey()), smallest.getValue());
+        } catch (NullPointerException e) {
+            return null;
+        }
+    }
+
+    public Entry<Customer, String> getNext(Customer customer) {
+        Set<Entry<Customer, String>> entryList = customerMap.entrySet();
+        Optional<Entry<Customer, String>> tmp = entryList.stream().filter(entry -> entry.getKey().getScores() > customer.getScores()).findFirst();
```

Comment:
По оформлению кода.
Лучше такое разбивать на строки:
```suggestion
        Optional<Entry<Customer, String>> tmp = entryList.stream()
                .filter(entry -> entry.getKey().getScores() > customer.getScores())
                .findFirst();
```